### PR TITLE
Refactor TEMPERATURE_FORMAT to use strong enum

### PR DESCRIPTION
**Pull request by sudo-michael**
Closes [#12400](https://github.com/OpenRCT2/OpenRCT2/issues/12400)  Refactor TEMPERATURE_FORMAT to use strong enum

### CODE REVIEW

File: src/openrct2/config/Config.h [hunk 1]
```diff
@@ -206,10 +208,10 @@ enum SORT
     SORT_DATE_DESCENDING,
 };
 
-enum TEMPERATURE_FORMAT
+enum class TEMPERATURE_FORMAT : int32_t
```
Comment (duncanspumpkin) [inline src/openrct2/config/Config.h:211]: Use CamelCase for enum classes.

File: src/openrct2/config/Config.h [hunk 2]
```diff
@@ -15,6 +15,8 @@
 
 #include <string>
 
+enum class TemperatureFormat : int32_t;
```
Comment (tupaschoal) [inline src/openrct2/config/Config.h:18]: Seems a bit overkill to have the underlying type as `int32_t` (even though that's what it was previously
```suggestion
enum class TemperatureFormat : bool;
```

I'd go for `bool`

Comment (duncanspumpkin) [reply src/openrct2/config/Config.h:18]: I think it was 32 due to it being a config option. There might be issues with changing the size.

Comment (tupaschoal) [reply src/openrct2/config/Config.h:18]: Compilation issues or running/replay issues? I'm alright if you want to keep it untouched for now

Comment (ZehMatt) [reply src/openrct2/config/Config.h:18]: Why would you pick a bool? That type has a different purpose and is not used for integers.

File: src/openrct2/config/Config.h [hunk 3]
```diff
@@ -206,10 +208,10 @@ enum SORT
     SORT_DATE_DESCENDING,
 };
 
-enum TEMPERATURE_FORMAT
+enum class TemperatureFormat : int32_t
```
Comment (ZehMatt) [inline src/openrct2/config/Config.h:211]:
```suggestion
enum class TemperatureUnit : int32_t
```

File: src/openrct2/config/Config.h [hunk 4]
```diff
@@ -206,10 +208,10 @@ enum SORT
...
 {
-    TEMPERATURE_FORMAT_C,
-    TEMPERATURE_FORMAT_F
+    C,
```
Comment (ZehMatt) [inline src/openrct2/config/Config.h:213]:
```suggestion
    Celcius,
```

File: src/openrct2/config/Config.h [hunk 5]
```diff
@@ -206,10 +208,10 @@ enum SORT
...
+    F
```
Comment (ZehMatt) [inline src/openrct2/config/Config.h:214]:
```suggestion
    Fahrenheit
```

File: src/openrct2/config/Config.h (hunk 2 above)
Comment (tupaschoal) [reply src/openrct2/config/Config.h]: Would fit just as well, unless we're planning to go for Kelvin eventually :¶

Not too attached, go for the the uint8 suggestion if it works

File: src/openrct2/config/Config.h (hunk 4 above)
Comment (tupaschoal) [reply src/openrct2/config/Config.h]: Celsius, no?

Comment (sudo-michael) [reply src/openrct2/config/Config.h]: Yup that's the correct spelling

Comment (ZehMatt) [reply src/openrct2/config/Config.h]: Yup my bad.

File: src/openrct2/config/Config.h (hunk 2 above)
Comment (ZehMatt) [reply src/openrct2/config/Config.h]: bool is just not logical for enums. I think int32_t will do fine.

**Pull request merged**